### ISSUE TITLE
Can't create an IotHub with a custom endpoint for storage

### ISSUE THREAD

**Issue by juniinacio** [ansible/ansible_collections_azure#1126]
##### SUMMARY

I'm trying to create an IotHub with an storage endpoint using the module `azure.azcollection.azure_rm_iothub`, but receive a very strange cryptic error, as I'm not that experienced with Python error messages and also not that experienced with Ansible, is very difficult for me to figure out what is exactly going wrong.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure.azcollection.azure_rm_iothub

##### ANSIBLE VERSION

```paste below
ansible [core 2.14.3]
  config file = /opt/iothub-demo/ansible.cfg
  configured module search path = ['/home/hass/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/hass/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/hass/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/hass/.local/bin/ansible
  python version = 3.10.7 (main, Mar 10 2023, 10:47:39) [GCC 12.2.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /home/hass/.local/lib/python3.10/site-packages/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.14.0

# /home/hass/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.14.0
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /opt/iothub-demo/ansible.cfg
DEFAULT_HOST_LIST(/opt/iothub-demo/ansible.cfg) = ['/opt/iothub-demo/inventory/hosts']
DEFAULT_VAULT_PASSWORD_FILE(/opt/iothub-demo/ansible.cfg) = /opt/iothub-demo/.vault_pass
HOST_KEY_CHECKING(/opt/iothub-demo/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Azure Cloud

##### STEPS TO REPRODUCE

Create the following playbook and try to create an IotHub.

```yaml
---
  - name: IoTHub demo
    hosts: localhost
    connection: local

    environment:
      AZURE_CLIENT_ID: "{{azure_application_id}}"
      AZURE_SECRET: "{{azure_application_secret}}"
      AZURE_SUBSCRIPTION_ID: "{{azure_subscription_id}}"
      AZURE_TENANT: "{{azure_tenant_id}}"
      ANSIBLE_AZURE_AUTH_SOURCE: "env"

    tasks:

       - name: Create Resource Group
         azure.azcollection.azure_rm_resourcegroup:
           name: "{{azure_resource_group}}"
           location: "{{azure_location}}"
           auth_source: env

      - name: Gathering Storage Account Facts
        azure.azcollection.azure_rm_storageaccount_info:
          resource_group: "{{azure_resource_group}}"
          name: "{{azure_iot_hub_storage_account_name}}"
          show_connection_string: true
        register: storage_account_output

      - name: "Set Storage Account Connection String"
        set_fact:
          storage_account_connection_string: "{{storage_account_output.storageaccounts[0].primary_endpoints.blob.connectionstring}}"

      - name: Create IoT Hub Storage Endpoint
        azure.azcollection.azure_rm_iothub:
          resource_group: "{{azure_resource_group}}"
          name: "{{azure_iot_hub_name}}"
          sku: "{{azure_iot_hub_sku}}"
          routing_endpoints:
            - name: "$default"
              connection_string: "{{ storage_account_connection_string }}"
              container: "iothub"
              encoding: "json"
              resource_group: "{{azure_resource_group}}"
              resource_type: "storage"
```

##### EXPECTED RESULTS

The module creates an IotHub with an custom connection to the configured storage account.

##### ACTUAL RESULTS

```paste below
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: RoutingStorageContainerProperties.__init__() missing 1 required keyword-only argument: 'container_name'
fatal: [localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/hass/.ansible/tmp/ansible-tmp-1679640282.7247028-415735-177694549729349/AnsiballZ_azure_rm_iothub.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/home/hass/.ansible/tmp/ansible-tmp-1679640282.7247028-415735-177694549729349/AnsiballZ_azure_rm_iothub.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/hass/.ansible/tmp/ansible-tmp-1679640282.7247028-415735-177694549729349/AnsiballZ_azure_rm_iothub.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_iothub', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_iothub', _modlib_path=modlib_path),\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iothub.py\", line 892, in <module>\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iothub.py\", line 888, in main\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iothub.py\", line 598, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 472, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iothub.py\", line 688, in exec_module\n  File \"/tmp/ansible_azure.azcollection.azure_rm_iothub_payload_n4yvudu_/ansible_azure.azcollection.azure_rm_iothub_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_iothub.py\", line 802, in construct_routing_endpoint\nTypeError: RoutingStorageContainerProperties.__init__() missing 1 required keyword-only argument: 'container_name'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Comment by juniinacio** [comment]
I see I pasted the wrong name for the routing endpoint, it should have been "storage".

```
- name: Create IoT Hub Storage Endpoint
  azure.azcollection.azure_rm_iothub:
    resource_group: "{{azure_resource_group}}"
    name: "{{azure_iot_hub_name}}"
    sku: "{{azure_iot_hub_sku}}"
    routing_endpoints:
      - name: "storage"
        connection_string: "{{ storage_account_connection_string }}"
        container: "iothub"
        encoding: "json"
        resource_group: "{{azure_resource_group}}"
        resource_type: "storage"
```

**Comment by Fred-sun** [comment]
@juniinacio Thank you for reporting this problem. I can also repeat this problem, and I am trying to solve it. Thank you very much!

**Comment by juniinacio** [comment]
> @juniinacio Thank you for reporting this problem. I can also repeat this problem, and I am trying to solve it. Thank you very much!

You are very welcome. You know one thing I also have noticed and was wondering, is that there are integration tests for the module, but they don't cover every use case of the module for example, they only cover creating a basic IotHub but not underlying pieces, is that by design or do you guys have a reason for not doing that? Maybe time consuming? 

But In either case very glad to have people like you that spent time to help make other peoples lives easier.

**Comment by Fred-sun** [comment]
Fixed by #1763 
```e.g.
Sample as:
    - name: create resource group
      azure_rm_resourcegroup:
        name: "{{ resource_group }}"
        location: eastus

    - name: Create storage account with static website enabled
      azure_rm_storageaccount:
        resource_group: "{{ resource_group }}"
        name: "{{azure_iot_hub_storage_account_name}}"
        account_type: Standard_LRS
        kind: StorageV2
        public_network_access: Enabled
        static_website:
          enabled: true
          index_document: "abc.htm"

    - name: Gathering Storage Account Facts
      azure_rm_storageaccount_info:
        resource_group: "{{resource_group}}"
        name: "{{azure_iot_hub_storage_account_name}}"
        show_connection_string: true
      register: storage_account_output

    - name: Create container
      azure_rm_storageblob:
        resource_group: "{{ resource_group }}"
        account_name: "{{azure_iot_hub_storage_account_name}}"
        container_name: my-blobs

    - name: "Set Storage Account Connection String"
      set_fact:
        storage_account_connection_string: "{{storage_account_output.storageaccounts[0].primary_endpoints.blob.connectionstring}}"

    - name: Create IoT Hub Storage Endpoint
      azure_rm_iothub:
        resource_group: "{{resource_group}}"
        name: "{{azure_iot_hub_name}}"
        sku: "{{azure_iot_hub_sku}}"
        routing_endpoints:
          - name: "iothub"
            connection_string: "{{ storage_account_connection_string }}"
            resource_type: storage
            container: "my-blobs"
            encoding: "json"
            resource_group: "{{resource_group}}"
